### PR TITLE
fix(catalogs): custom namespace catalog build fail

### LINKED ISSUE #218: [BUG] oc-mirror fails when mirroring operator catalogs using a custom namespace
# Version

```console
$ oc-mirror version
v0.1.0-alpha.3
```

# What happened?

Recieved error ` MANIFEST_UNKNOWN: manifest unknown` during the catalog rebuild portion of publishing operations.

# What you expected to happen?

Catalog to rebuild successfully within the top-level namespace

# How to reproduce it (as minimally and precisely as possible)?

```console
$ oc-mirror --config catalog.yaml docker://localhost:5000/myns
```

# Anything else we need to know?

# References

### PR DESCRIPTION
**Pull request by jpower432**
# Description

Fixes #218 

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [X] Update E2E test

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pkg/cli/mirror/catalog_images.go [hunk 1]
```diff
@@ -74,8 +74,10 @@ func (o *MirrorOptions) rebuildCatalogs(ctx context.Context, dstDir string, file
 				return fmt.Errorf("error parsing index dir path %q as image %q: %v", fpath, img, err)
 			}
 			// Update registry so the existing catalog image can be pulled.
-			// QUESTION(estroz): is assuming an image is present in a repo with the same name valid?
 			ctlgRef.Ref.Registry = mirrorRef.Ref.Registry
+			if len(o.UserNamespace) != 0 {
+				ctlgRef.Ref.Namespace = strings.Join([]string{o.UserNamespace, ctlgRef.Ref.Namespace}, "/")
```
Comment (estroz) [inline pkg/cli/mirror/catalog_images.go:79]:
```suggestion
				ctlgRef.Ref.Namespace = path.Join(o.UserNamespace, ctlgRef.Ref.Namespace)
```

File: pkg/cli/mirror/catalog_images.go [hunk 2]
```diff
@@ -161,9 +163,11 @@ func (o *MirrorOptions) rebuildCatalogs(ctx context.Context, dstDir string, file
 			if err != nil {
 				return refs, fmt.Errorf("error parsing image %q: %v", OPMImage, err)
 			}
-			// Update registry so the existing OPM image can be pulled.
-			// QUESTION(jpower): is assuming an image is present in a repo with the same name valid?
+
 			opmImage.Registry = mirrorRef.Ref.Registry
+			if len(o.UserNamespace) != 0 {
+				opmImage.Namespace = strings.Join([]string{o.UserNamespace, opmImage.Namespace}, "/")
```
Comment (estroz) [inline pkg/cli/mirror/catalog_images.go:169]: Same thing here

Comment (jpower432) [reply pkg/cli/mirror/catalog_images.go:169]: Gotcha. I need to make that change in a few places. Working on that now.

Comment (jpower432) [reply pkg/cli/mirror/catalog_images.go:169]: Gotcha. I need to make that change in a few places. Working on that now.

**Pull request merged**